pkg/tfpfbridge/provider_resources.go

```diff
@@ -27,15 +27,19 @@ import (
 	"github.com/pulumi/pulumi-terraform-bridge/pkg/tfpfbridge/pfutils"
 )
 
-func (p *Provider) resources(ctx context.Context) (res pfutils.Resources, err error) {
+type resourceHandle struct {
+	makeResource          func() pfresource.Resource
+	terraformResourceName string
+	schema                tfsdk.Schema
+	pulumiResourceInfo    *info.ResourceInfo // optional
+	idExtractor           idExtractor
```

Comment:
I'm curious why this is using functions as struct fields instead of creating methods on the `resourceHandle` type? is it because each resource handle needs to have its own unique extractor, based on the resource type?

Reply:
Ah these are some bad habits from my FP past. It's true that each idExtractor is slightly different but the only difference really is knowing the resource type for prettier errors. Moving this to a method and its state into the resourceHandle struct makes sense.